mesa/experimental/cell_space.py

```diff
@@ -0,0 +1,204 @@
+import itertools
+import random
+from collections.abc import Iterable
+from functools import cache, cached_property
+from typing import Callable, Optional
+
+from .. import Agent
+
+Coordinate = tuple[int, int]
+
+
+class Cell:
+    __slots__ = ["coordinate", "_connections", "agents", "capacity", "properties"]
+
+    def __init__(self, i: int, j: int, capacity: int | None = 1) -> None:
+        self.coordinate = (i, j)
+        self._connections: list[Cell] = []
+        self.agents: list[Agent] = []
+        self.capacity = capacity
+        self.properties: dict[str, object] = {}
+
+    def connect(self, other) -> None:
+        """Connects this cell to another cell."""
+        self._connections.append(other)
+
+    def disconnect(self, other) -> None:
+        """Disconnects this cell from another cell."""
+        self._connections.remove(other)
+
+    def add_agent(self, agent: Agent) -> None:
+        """Adds an agent to the cell."""
+        if self.capacity and len(self.agents) >= self.capacity:
+            raise Exception("ERROR: Cell is full")
+        if isinstance(agent.cell, Cell):
+            agent.cell.remove_agent(agent)
+        self.agents.append(agent)
+        agent.cell = self
+
+    def remove_agent(self, agent: Agent) -> None:
+        """Removes an agent from the cell."""
+        self.agents.remove(agent)
+        agent.cell = None
+
+    @property
+    def is_empty(self) -> bool:
+        """Returns a bool of the contents of a cell."""
+        return len(self.agents) == 0
+
+    @property
+    def is_full(self) -> bool:
+        """Returns a bool of the contents of a cell."""
+        return len(self.agents) == self.capacity
+
+    def __repr__(self):
+        return f"Cell({self.coordinate}, {self.agents})"
+
+    @cache
+    def neighborhood(self, radius=1, include_center=False):
+        if radius == 0:
+            return {self: self.agents}
```

Comment:
So this is just returning agents at the current location? But if include_center=False, that's conceptually incorrect right?

So should we allow it at all? I think for clearness I would be in favor of not allowing radius==0 and making a separate function for it.

Reply:
Ah I understand, that ends the loop on line 67-68.

Smart, but I'm not sure I fully like it.

Reply:
Exactly, its for ending the recursion. I would say radius=0 and include_self=False dont make sense together. Actually radius=0 already doesnt make sense from a user perspective, but is used for the recursion

Reply:
I think it is possible to get rid of the radius=1 check but instead raise an exception if radius < 1. I think I might even have the code in that way in my work from last weekend.

Reply:
What will happen with float radius in the continiousspace? We override the method?

Reply:
We haven't thought about continuous space yet. All this really only works for discrete spaces. So I suggest we don't worry about that yet.